bee-ledger/src/workers/snapshot/import.rs

```diff
@@ -282,32 +278,20 @@ pub(crate) async fn import_snapshots<B: StorageBackend>(
 
     if !full_exists && delta_exists {
         return Err(Error::Snapshot(SnapshotError::OnlyDeltaSnapshotFileExists));
-    }
-
-    if !full_exists {
-        download_snapshot_file(
+    } else if !full_exists && !delta_exists {
+        download_latest_snapshot_files(
+            network_id,
             config.full_path(),
-            config.download_urls().iter().map(DownloadUrls::full),
+            config.delta_path(),
+            config.download_urls(),
         )
         .await?;
     }
 
-    // Full snapshot file exists from now on.
-
     import_full_snapshot(storage, config.full_path(), network_id).await?;
 
-    if let Some(delta_path) = config.delta_path() {
-        if !delta_exists
-            && download_snapshot_file(delta_path, config.download_urls().iter().map(DownloadUrls::delta))
-                .await
-                .is_err()
-        {
-            warn!("Could not download the delta snapshot file and it will not be imported.");
-        }
-
-        if delta_path.exists() {
-            import_delta_snapshot(storage, delta_path, network_id).await?;
-        }
+    if config.delta_path().map(Path::exists).is_some() {
```

Comment:
Is the map actually useful here ? We only evaluate the Some/None but never the content of the Some ?

Reply:
`Path::exists` should actually be analysed to decide whether to import or not

Reply:
`if let Some(true) = config.delta_path().and_then(Path::exists)` something like that ?

Reply:
You are right, that's a bad one. 🙈